codyze-core/src/main/kotlin/de/fraunhofer/aisec/codyze/CodyzeScriptCompilationConfiguration.kt

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright (c) 2025, Fraunhofer AISEC. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *                    $$$$$$\  $$$$$$$\   $$$$$$\
+ *                   $$  __$$\ $$  __$$\ $$  __$$\
+ *                   $$ /  \__|$$ |  $$ |$$ /  \__|
+ *                   $$ |      $$$$$$$  |$$ |$$$$\
+ *                   $$ |      $$  ____/ $$ |\_$$ |
+ *                   $$ |  $$\ $$ |      $$ |  $$ |
+ *                   \$$$$$   |$$ |      \$$$$$   |
+ *                    \______/ \__|       \______/
+ *
+ */
+package de.fraunhofer.aisec.codyze
+
+import de.fraunhofer.aisec.codyze.dsl.Import
+import java.io.File
+import java.net.JarURLConnection
+import java.net.URL
+import kotlin.script.experimental.annotations.KotlinScript
+import kotlin.script.experimental.api.*
+import kotlin.script.experimental.dependencies.CompoundDependenciesResolver
+import kotlin.script.experimental.dependencies.ExternalDependenciesResolver
+import kotlin.script.experimental.dependencies.FileSystemDependenciesResolver
+import kotlin.script.experimental.host.FileBasedScriptSource
+import kotlin.script.experimental.host.FileScriptSource
+import kotlin.script.experimental.jvm.dependenciesFromClassContext
+import kotlin.script.experimental.jvm.jvm
+import kotlin.script.experimental.util.filterByAnnotationType
+import kotlin.script.templates.ScriptTemplateDefinition
+
+/**
+ * Abstract base class for Codyze scripts. Codyze scripts are Kotlin scripts that can be used to
+ * - Define a project structure that describes the "target of evaluation" (ToE) for the analysis as
+ *   well as requirements the ToE must fulfill
+ * - Define queries that evaluate whether the requirements are met
+ *
+ * This class is the scription definition (template) needed for the Kotlin compiler to recognize
+ * this as a script.
+ */
+@ScriptTemplateDefinition(scriptFilePattern = ".*\\.codyze\\.kts")
+@KotlinScript(
+    // File extension for the script type
+    fileExtension = "codyze.kts",
+    // Compilation configuration for the script type
+    compilationConfiguration = CodyzeScriptCompilationConfiguration::class,
+    // Evaluation configuration for the script type
+    // evaluationConfiguration = CodyzeScriptEvaluationConfiguration::class,
+)
+abstract class CodyzeScript
+
+val baseLibraries =
```

Comment:
It would be nice to have few comments in this file even if there's probably not much code that will ever be touched